---
bibliography: 01_introduction/references.bib
---

# Introduction

<!-- TODO: add figure number references -->
<!-- TODO: add section label references -->
<!-- TODO: update with new findings since June 2013 -->
<!-- TODO: photosynthesis summary, importance. -->

C<sub>4</sub> photosynthesis (hereafter C<sub>4</sub>) has evolved independently at least 66 times to solve the problem of the wasteful oxygenation reaction catalysed by RuBisCO [@sage2011]. Despite accounting for only a small fraction of all plant species, C<sub>4</sub> plants contribute disproportionately to terrestrial carbon fixation [@lloyd1994], dominate many tropical and subtropical habitats [@cerling1997; @osborne2006], and have become even more widely successful through their overrepresentation in human agriculture [@brown1999].

In the first part of this introductory chapter, C<sub>4</sub> photosynthesis is placed in its physiological and evolutionary context. The humanitarian importance of understanding C<sub>4</sub> is explained, after which the current knowledge of the development and control of C<sub>4</sub> is reviewed and the further discoveries necessary to bring this knowledge to application are outlined.

The second part of the chapter explains the theoretical underpinnings and practical process of <i>de-novo</i> transcriptomics, an experimental approach being fruitfully harnessed in the pursuit of understanding C<sub>4</sub>. The principles and merits of this approach are discussed, before the potential insights and limitations are summarised.

The third and final part of this chapter reviews the various ways in which transcriptomics has been harnessed in C<sub>4</sub> research. The insights that have been gleaned are discussed, and the potential for further inference from existing data is highlighted. A roadmap for tool and resource development is proposed that would allow considerably more informative inference from existing data, and ultimately underpin progress towards the goal of understanding and engineering C<sub>4</sub>.

## C<sub>4</sub> photosynthesis

### C<sub>4</sub> solves an ancient problem that affects almost all life on earth

The light-independent reactions of photosynthesis are highly conserved @kawashima1970 and involve CO<sub>2</sub> being fixed with ribulose-1,5-bisphosphate (RuBP) by the enzyme ribulose-1,5-bisphosphate carboxylase oxygenase (RuBisCO) to form two molecules of 3-phosphoglycerate (G3P). One of these G3P molecules re-enters the Calvin Benson cycle while the other is incorporated into a sucrose molecule (figure \ref{fig:calvincycle}). Sugars produced in this way are by far the largest primary energy source in the biosphere [@fuchs2011]; RuBisCO is thus the major conduit through which life on Earth is energised.

![Photorespiration and the Calvin Benson cycle. Grey circles represent enzymes; grey lines indicate reactions of photorespiration that produce wasteful or harmful byproducts. Adapted from @peterhansel2010](figures/photorespiration.png) {#fig:photorespiration}

RuBisCO can also catalyse the reaction of molecular oxygen (\otwo) with RuBP, forming one G3P and one 2-phosphoglycolate (2PG). While RuBisCO has a specificity for CO<sub>2</sub> over \otwo of ~80$\times$ [@jordan1981] oxygen is ~530$\times$ more adundant in the atmosphere, so that approximately a quarter of RuBisCO-catalysed reactions are oygenations [@ehleringer1991]. This oxygenation reaction is wasteful for several reasons. 2PG is an inhibitor of several enzymes linked to the Calvin cycle [@anderson1971; @kelly1976] and must therefore be de-phosphorylated to produce glycolate. Glycolate decomposes rapidly to glyoxylate which inhibits activation of RuBisCO [@chastain1989; @campbell1990]. To avoid metabolic retardation, the the conversion of glyoxylate to the safe end-product glycine is required, which releases CO<sub>2</sub> (thus revoking the initial carbon fixation) and NH<sub>3</sub> (which is toxic). In order to recover a sugar product, glycine is converted in multiple steps to a single G3P, resulting in a total net cost of ~12.25 ATP for a single round of the entire photorespiratory pathway [@peterhansel2010], and a net loss of CO<sub>2</sub>. As a result of this inefficiency, C<sub>4</sub> plants must produce abundant RuBisCO to achieve necessary rates of carbon fixation, leading to RuBisCO being probably the most abundant protein on earth [@ellis1979; @raven2013], and a considerable burden on net photosynthesis in most plants [@sharkey1988].

By fixing CO<sub>2</sub> through \ce{HCO3-} and a succession of 4-carbon molecules [@hatch1966] in mesophyll (M) cells, and shuttling the resultant four-carbon molecule into bundle sheath (BS) cells with high RuBisCO content where CO<sub>2</sub> is released, C<sub>4</sub> plants are able to provide a CO<sub>2</sub>-rich environment for RuBisCO [@furbank1987]; see figure \ref{fig:c4biochem}). This reduces the oxygenase activity of RuBisCO and avoids wasteful photorespiration, resulting in greater total energy capture from photosynthesis.

![C<sub>4</sub> photosynthesis compartmentalises the light-independent reactions of photosynthesis. CO<sub>2</sub> is fixed initially in mesophyll cells, resulting in a 4-carbon acid that diffuses into the bundle sheath cells, where RuBisCO is concentrated. The 4-carbon acid is decarboxylated in the vicinity of RuBisCO, recreating the high-CO<sub>2</sub> environment in which the enzyme evolved. This greatly reduces the rate of photooxidation, in turn reducing the need for photorespiration and improving overall efficiency of carbon capture.](figures/c4biochem.png) {#fig:c4_biochem}

Because less RuBisCO is required in C<sub>4</sub> plants, they tend to have much higher nitrogen use efficiency [@brown1978], and due to the CO<sub>2</sub> concentration mechanism a lower stomatal conductance is possible, resulting in greater water use efficiency [@bjorkman1971; @brown1999]. These factors affect radiation use efficiency (RUE), and therefore total potential yield, which is on average around 50\% greater in C<sub>4</sub> than C<sub>3</sub> plants [@monteith1978; @sheehy2008].

C<sub>4</sub> photosynthesis is typically associated with a suite of metabolic, anatomical and ultrastructural specialisations known as Kranz anatomy [@dengler1999]. These adaptations include: metabolic compartmentalisation, such that certain genes are expressed preferentially in either BS or M cells; increased density of BS cells within the leaf, such that each BS cell is usually surrounded by a single layer of M cells, with a concomitant increased density of leaf veins; presence of a suberin layer in BS cell walls to prevent CO<sub>2</sub> leakage (in NADP types); increased density of plasmodesmata linking BS and M cells to allow increased metabolite transport; and BS chloroplasts that are partitioned centrifugally (in most examples of NADP-ME subtype) or centripetally (in most examples of NAD-ME subtype).

<!-- % TODO: go into more detail about C4 biochemistry here and how it was discovered -->

### The profligate evolution of C<sub>4</sub> photosynthesis

Oxygenic photosynthesis using RuBisCO probably arose 2.9 billion years ago (Ga) and was abundant by 2.6Ga [@nisbet2007]. This early atmosphere was rich in CO<sub>2</sub> and low in \otwo relative to modern levels [@bekker2004; @holland2006], so RuBisCO's oxygenation reaction was not a constraint for early photosynthetic organisms. Over the next two billion years, first cyanobacteria, then algae and land plants, gradually oxygenated the atmosphere from ~0.04atm 2.4Ga to ~0.2atm 0.5Ga, close to modern levels [@holland2006]. At the same time, CO<sub>2</sub> levels were depleted from ~2atm 4.5Ga to ~0.001atm 0.5Ga [@kasting1987] due to cyanobacterial and algal photosyntheis. Atmospheric CO<sub>2</sub> then underwent two periods of sudden decline in the last 0.5Ga: firstly ~300Ma [@berner1998] due to the rise of land plants, and secondly ~30Ma in the Oligocene epoch [@pagani2005; @beerling2011].

<!-- repeated evolution -->

C<sub>4</sub> photosynthesis arose for the first time during, and probably in response to, the Oligocene CO<sub>2</sub> minimum [@vicentini2008; @christin2008], and did so multiple times independently in phylogenetically diverse lineages. The earliest origins occured in the Poales and Caryophyllales [@christin2011], which are now the most diverse C<sub>4</sub> clades, each containing at least 23 putative independent C<sub>4</sub> origins [@sage2011]. In total, C<sub>4</sub> is found in at least 66 independent plant lineages whose common ancestors at C<sub>3</sub> and are thus thought to represent separate evolutionary origins [@grassphylogenyworkinggroupii2012; @sage2011-b].

<!-- C4 uses components which already existed in C3 ancestors. They performed different functions, and have been co-opted -->

The repeated evolution of C<sub>4</sub> may have been facilitated by the fact that the key components of the C<sub>4</sub> apparatus already existed in C<sub>3</sub> ancestors. The structures that became specialised for compartmentalisation of metabolism (BS and M cells), inter-cellular flux of metabolites (plasmodesmata) and CO<sub>2</sub> concentration maintenance (BS cell walls) were present ancestrally and needed only slight modification during development to serve their C<sub>4</sub> purpose [@christin2013]. Similarly, no new enzymes needed to evolve for the pathway to operate; all were co-opted through duplication with neofunctionalisation from various roles in the ancestral C<sub>3</sub> species, particularly from general housekeeping roles @monson2003, [@aubry2011]. The key evolutionary innovations in C<sub>4</sub> involved modification of the expression patterns of the co-opted enzymes [@hibberd2010; @williams2012], possibly driven by parallel evolution of regulatory transcription factors (Aubry et al., in prep.).

<!-- TODO: cite Aubry et al now published -->

<!-- some aspects of C4 development appear to be preadaptations -->

## The humanitarian case for crop engineering interventions

Rice is the staple food for more people than any other crop, providing approximately 8\% of the total calories consumed by humans globally  (figure \ref{fig:agriculturedata}A) and around 20\% of protein. The proportional contribution of rice to the human diet has been steadily increasing over the last 50 years (figure @fig:agriculturedata A).

The global population passed the 7 billion mark in 2012 and continues to rise, projected to reach 9.3 billion by 2050 [@undesa2010]; figure \ref{fig:agriculturedata}B). Developing countries account for a disproportionately high share of population growth. To avoid increases in hunger and poverty, food production must rise to meet the needs of the increasing population, and in particular developing economies in Asia and Africa must be able to meet the demands of their populations. Contemporarily, poverty and hunger are widespread, thus further increases in productivity will be needed if current problems are to be addressed, in addition to preventing future ones.

![Global agricultural and consumption trends demonstrate the need for crop improvement intervention. A.) Area harvested for rice 1961-2010. Increases in area harvested for rice accounted for a large proportion of the increases in total rice production which helped meet the needs of growing populations from 1960 to the 1990s. Around 2000 the area under rice cultivation reached the maximum land available (produced using data from http://faostat3.fao.org/). B.) Past and projected global and Asian Developing Nations (ADN) populations (produced using data from @undesa2010)](figures/agriculturedata.png) {#fig:agriculturedata}

Throughout and since the Green Revolution the rate of yield increase has remained steady. However, the proportional contribution of breeding to that increase has steadily declined, while the proportion of yield increase derived from increased fertilizer use has risen [@zhongyu2000]. This presents a major challenge to food production; increases from intensification of fertiliser are predicted to abate in the near future, and problems of sustainability threaten continued reliance on nutrient enrichment of soils.

The area of land under production of rice increased greatly from 1960-1990, and this increase contributed to the ability of the rice supply to keep up with the increase in demand. However, from 1990-2010 increases in area under production have fallen as the limits of arable land area are reached (figure @fig:agriculturedata A). Thus in the future, unless new rice strains are developed which can enable more marginal lands to be brought under production, increases in area farmed will contribute negligibly to increases in rice production.

Taken together, these factors demonstrate a clear and urgent need for greater increases in rice yields, area under production, or both. Various technical solutions have been proposed, many of which include improving tolerance of rice to abiotic stresses such as drought, or involve incremental increases in yield. However, it is now increasingly recognised that the yield of modern rice strains is close to the theoretical maximum for the plant [@evenson2003]. Thus an intervention is needed that can raise the theoretical maximum yield. By introducing a different, more efficient photosynthetic system into rice, maximum yield capacity could be greatly increased at the same time as decreasing water and nitrogen requirements.

## The C<sub>4</sub> rice project

A coordinated international effort is underway to engineer the C4 photosynthetic machinery into rice [@hibberd2008]. This program, and similar programs for other C3 crops (notably wheat), could produce a vast humanitarian benefit by increasing the maximum yield potential of some of the worldâ€™s major staple food crops by approximately 50\% [@mitchell2006].

Thus far, individual enzymes of the pathway from maize, which is very well characterised, have been individually transformed into a rice background and are now being stacked by crossing [@kajala2011]. Mutant screens have been carried out seeking determinants of kranz anatomy in sorghum and maize, or precursory elements of kranz in rice, but have yielded few actionable insights [@langdale2011]. Now, systems-level approaches approaches being used in existing and new model species to attempt to elucidate the regulatory elements of the pathway.

## The developmental dynamics of the C<sub>4</sub> leaf

C<sub>4</sub> leaves must execute a developmental programme that enables the differentiation of BS and M cells anatomically, ultrastructurally and biochemically. All C<sub>4</sub> leaves begin their development with C<sub>3</sub>-like anatomy and physiology, and the C<sub>4</sub> state is gradually overlaid on the ancestral state by developmental regulation. In maize and *Amaranth*, the partitioning of photosynthetic genes into their respective cell types depends on the developmental light cues the leaf receives [@langdale1988; @wang1993], and in maze this programme can only be initiated in cells immediately surrounding the vasculature [@langdale1991]. In maize [@majeran2005; @majeran2009] and C<sub>4</sub> *Sueda* [@koteyeva2011], *Flaveria* (@mckown2007, Kuempers, pers.comm) and *Cleome* (Aubry, pers. comm.) species, vasculature and plastid morphology and arrangement gradually reach the mature C<sub>4</sub> state along a developmental gradient in young C<sub>4</sub> leaves, with increasing density of venation and differential chloroplast structure between BS and M cells being representative of the transition.

<!-- TODO: replace pers comm in previous paragraph with citations of subsequently published work -->

Understanding the molecular basis of this developmental programme is of primary importance, and the C<sub>4</sub> community has converged on high-throughput methods of molecular quantification as a key and fruitful approach. In maize, quantitative proteomics [@majeran2005; @majeran2008; @majeran2010], microarray analysis [@sawers2007], and recently RNA-seq [@li2010-b; @chang2012; @wang2013; @liu2013] have been leveraged in experiments designed to capture cell-type specific and/or developmental differences in peptide and transcript expression. These experiments have enabled the identification of candidate transcription factors and other regulatory elements whose expression coincides with C<sub>4</sub> genes, and are therefore candidate C<sub>4</sub> regulators which are now being characterised [@covshoff2012].

While these detailed analyses have so far been restricted to maize due to the extensive genomic resources available in that system, recent advances in high-throughput sequencing, <i>de-novo</i> assembly, expression quantification (see section \ref{sec:transcriptomics}) and cell-specific RNA extraction (e.g. @covshoff2013) now enable the application of developmental and M/BS specific high-throughput analysis in a wider range of non-model species. Such experiments have been recently completed in *Cleome* (Aubry et al, in press) and are currently underway in *Flaveria* (Kuempers, pers. comm.) and *Setaria* (John, pers. comm.). The repeated evolution of C<sub>4</sub> necessitates its study in a broad range of species, both to give insight into the evolution of the syndrome at the molecular level, and to discover whether the regulatory mechanisms underlying development and maintenance C<sub>4</sub> evolved in parallel.

<!-- TODO: replace pers comm in previous paragraph with citations of subsequently published work -->

A complementary experimental design in which expression profiles are compared between closely [@gowik2011; @brautigam2011] or distantly (Aubry et al, in press) related C<sub>4</sub> and C<sub>3</sub> species enables those regulatory elements which are exclusively associated with C<sub>4</sub>, as opposed to having correlated expression by chance, to be identified. The ongoing 1000 Plant Transcriptomes Project (www.onekp.com) is sequencing, assembling and quantifying transcriptomes for 26 C<sub>3</sub> and 39 C<sub>4</sub> sister species across an impressive phylogenetic range and encompassing two of the three distinct biochemical subtypes of C<sub>4</sub>. These will be analysed comparatively to identify enriched genes using a new bioinformatics pipeline developed for the project (Kelly et al., in press).

<!-- TODO: replace in press in previous paragraph with citations of subsequently published work -->


## Regulation in *cis* and *trans*

Control of genes recruited to C<sub>4</sub> can involve transcriptional and translational efficiency as well as post-transcriptional and post-translational degradation. These mechanisms can be targeted to promoters, coding regions or UTRs, and involve the evolution in both *cis* and *trans* factors.

*Cis*-elements in promoters of some C<sub>4</sub> genes are required for cell specificity. It has been shown that 1.7kb and 0.6kb regions upstream of *PEPC* in maize are sufficient for M-specific accumulation [@kausch2001], and that the cell specificity is not present in young leaves, suggesting developmental control. Additionally, a 41bp motif in the promoter of *PEPC* in C<sub>4</sub> *Flaveria spp.* with a G->A substitution and a CACT insertion [@akyildiz2007] has been shown to be necessary and sufficient for M-cell specificity [@gowik2004].

Nucleotide sequence in the coding region can generate cell-specific expression in C<sub>4</sub>. Accumulation of NAD-ME in *Cleome gynandra* BS cells requires a specific 240 nucleotide sequence to be present in the 5' region of the translated portion of the transcript [@brown2011]. Frame shift mutation does not affect cell specificity, showing that nucleotide rather than amino acid sequence is important. The orthologous *AtNAD-ME* gene also accumulated in BS cells in *C. gynandra* and again a 240 nucleotide sequence was found to be necessary, suggesting the presence in *C. gynandra* of a *trans*-factor responsible for BS localisation.

Cell specificity in some C<sub>4</sub>-recruited genes also relies on UTR sequence. In a functionally redundant manner, either 3' or 5' UTRs of $\beta$-carbonic anhydrase (*CA4*) and pyruvate, orthophosphate dikinase (*PPDK*) are sufficient for M-specific localisation in *C. gynandra* [@patel2004; @kajala2011]. UTRs of homologous genes from *A. thaliana* convey cell specificity when placed into *C. gynandra* [@kajala2011]. This again suggests *C. gynandra* has modifications in *trans* which are responsible for BS-cell specificity.

PEPC kinase (PEPCk), which phosphorylates PEPC in a light-dependent manner to regulate its activity [@nimmo2003], is rapidly degraded by the ubiquitin system in maize M cells [@agetsuma2005]. When the *ZmPEPC* gene is placed into rice the light-dark regulation is reversed [@fukayama2003]. This post-translational control of expression must depend on the coding sequence, but it is not clear whether this mechanism controls cell-specificity.

Williams (2012, submitted) found that the promoter, introns and UTRs of *CA4* from *C. gynandra* act as enhancer elements. In Arabidopsis, expression of *pCgCA4::uidA* (with 1025bp promoter) was 2-fold greater than *pAtCA4::uidA* (with 2119bp promoter) when measured using MU assays [@jefferson1987], demonstrating the role of the *CgCA4* promoter region. The *CgCA4* coding sequence with intact introns showed ~5-fold greater expression than the cDNA coding sequence when fused with *uidA* and placed under control of the 35S promoter in Arabidopsis, showing that introns of *CgCA4* act as enhancers. Bioinformatic analysis showed that introns in *CgCA4* were shorter on average than those in *AtCA4*. GUS/MU analysis of CgCA4 and AtCA4 with either or both of 5' and 3' UTRs missing showed that *CgCA4* UTRs had additive enhancer activity. The 5' *AtCA4* also showed enhancer activity when an intron in the UTR was removed, while the 3' UTR repressed expression. These results demonstrate that CA4 has complex regulation of expression mediated by multiple elements in *cis* and *trans*.

Synthesising these results, it is clear that the regulation of C<sub>4</sub> has involved evolution of regulatory elements in both *cis* and *trans*. *cis*-elements in promoters, UTRs, coding sequences and introns can direct localisation and magnitude of expression, while *trans*-factors interacting with those *cis*-elements must have been recruited in C<sub>4</sub> evolution.

Clearly, access to complete sequences of C<sub>4</sub> genes, transcripts and proteins in multiple species, as well as their expression in different tissues and an understanding of their contextual regulatory networks, is desirable.

## The importance of wetland C<sub>4</sub> monocots
\label{sec:wetlandc4}

Rain-fed paddy rice accounts for 90\% of rice production [@bouman2007], and a C<sub>4</sub> rice will therefore be a semi-aquatic C<sub>4</sub> plant. But previous and on-going comparative transcriptomic approaches have focused on either dicotyledons such as *Cleome* and *Flaveria* which are evolutionarily distant from cereals, or on C<sub>4</sub> monocots which grow in seasonally dry environments. It is therefore of importance, in the context of efforts to engineer C<sub>4</sub> rice, to understand how well C<sub>4</sub> might perform outside the hot and dry environment with which it is classically associated.

In fact, C<sub>4</sub> monocots in wetland environments are extremely productive. *Echinochloa polystachya*, a C<sub>4</sub> grass that grows semi-aquatically along the tropical Amazon floodplain, has been shown to be among the most productive plants in a natural setting @piedade1991, @piedade1994], achieving rates of carbon fixation equivalent to maize under intensive cultivation [@morison2000]. *Cyperus papyrus*, a wetland C<sub>4</sub> sedge growing in tropical African swamps, has been shown to have some of the highest rates of biomass accumulation of any plant in any habitat [@jones1988; @jones1997; @jones2002]. These impressive yields have been attributed to the high nitrogen use efficiency of C<sub>4</sub> plants, allowing them to thrive in the oligotrophic conditions in standing swamp water, in combination with the generally higher RUE provided by bypassing photorespiration. It is therefore clear that C<sub>4</sub> can perform extremely productively in a hot, wet environment.

As rice is cultivated in both tropical and temperate regions, it is equally important to consider whether C<sub>4</sub> can still perform well in cooler wet environments. Supporting evidence comes from C<sub>4</sub> weeds in rice agriculture. *Echinochloa glabrescens* often outcompetes rice in paddy cultivation [@rao1992], and has been shown to have a photosynthetic efficiency of 1.8$\times$ that of rice in field experiments [@sheehy2007], while completing its life cycle in 60\% of the time [@sage2011]. Thus there is precedent for fast-cropping, highly productive C<sub>4</sub> plants in the conditions used for rice production.

## Potential pitfalls of C<sub>4</sub> in the wet
\label{sec:wetpitfalls}

Knowing that C<sub>4</sub> can perform well in wetland conditions in certain species, a crucial question is whether the understanding of C<sub>4</sub> operation and regulation in current and emerging dry C<sub>4</sub> model species can be transferred to the wet environment. If it can, this provides a clear roadmap for transforming C<sub>4</sub> into rice. Conversely, if there are features that allow C<sub>4</sub> to be productive in the wet that are unique to wetland species, those features must be understood in order that the knowledge can be translated to a C<sub>4</sub> rice.

It seems likely that, given the unique physico-chemical conditions in semi-aquatic environments, there will be at least some such insights to be gained by studying wetland-adapted C<sub>4</sub> species. Exemplifying this, ammonium is the major nitrogen source in waterlogged soils and wetlands [@watanabe1986], and requires special metabolic machinery to be of use in the plant [@lam1996]. @masumoto2010 discovered that one isoform of PEPC in rice, *Osppc4*, is localised to the mesophyll chloroplast, and that this localisation is essential for successful nitrogen acquisition from ammonium. Previous attempts to overexpress C<sub>4</sub> enzymes in rice showed that, when PEPC and malic enzyme (ME) are expressed together in high concentrations, plants become stunted, possibly due to a futile metabolic cycling between the two [@taniguchi2008; @miyao2011]. However, in C<sub>4</sub> plants ME is chloroplast localised, suggesting that in combination with the chloroplast-specific *Osppc4*, a futile cycle could thwart attempts to place C<sub>4</sub> enzymes in the appropriate compartments in rice. This highlights the importance of understanding how C<sub>4</sub> operates in a wetland species to identify how the ammonium assimilation problem is solved, and whether there are other similar issues that need to be considered.

## *Echinochloa glabrescens* as a model for wet C<sub>4</sub>

*E. glabrescens* grows alongside rice [@rao1992], outcompetes it unless rigorously suppressed [@kim1980], and has a nearly identical growth form [@kim1989]. These facts suggest that insights gained from studying the operation of C<sub>4</sub> in *E. glabrescens* will be directly translatable to engineering rice. In addition, a considerable literature exists on the species due to its interest as a pest of rice cultivation (e.g. @janiya1984, @kim1987, @kim1988, @alberto1996). It is therefore suggested that *E. glabrescens* be developed as system for studying C<sub>4</sub> photosynthesis in the wet.


## <i>de-novo</i> transcriptomics
\label{sec:transcriptomics}

The transcriptome of a tissue is the population of mRNAs present at a snapshot in time. <i>de-novo</i> transcriptomics involves assaying the transcriptome of a species which does not have a sequenced genome, and requires using only the evidence present in sequenced reads to reassemble the sequence of origin. In principle, <i>de-novo</i> transcriptomics promises massively parallel retrieval of the sequences of all transcripts present in a sample, along with highly sensitive estimates of the abundance of those transcripts. In practise however, the ability to do this is dependent on the interface between DNA sequencing technology, computational tools, and cost in both time and money.

Because <i>de-novo</i> transcriptomics offers the potential to obtain the sequences of many transcripts, it is by far the most cost-effective method for investigating multiple gene sequences in a novel species. And because expression quantification can be performed using the same data, systems-level approaches to reconstructing regulatory networks are made possible. In general, expression quantification with a high degree of accuracy, as measured by comparison to qPCR data, is possible using current methods. Accurate reconstruction of full-length transcripts for the purposes of nucleotide-sensitive phylogenetic analysis or motif discovery, however, is not yet mature.

Although the availability of computing power continues to increase rapidly, the capacity of sequencing technology is growing even faster. As a result, the running time of the algorithms used to perform simple overlap consensus assemblies such as the first draft of the human genome have become unworkable with modern high-throughput sequencing data [@stein2010]. Errors and uncertainties in the sequence data complicate the process and need to be addressed prior to assembly. Simple approaches to quality control of sequence data include trimming low-quality bases from reads and removing reads with an average quality below some threshold [@schmieder2011; @lohse2012]. To further remove errors which are not identified by the quality scores produced by the sequencing machine, various models exist which attempt to learn the locations of errors by comparing reads to one another [@le2013], or using graph-based metrics [@brown2012]. To perfectly reassamble the sequences of transcripts <i>de-novo</i> using even error-corrected short reads is a challenge far beyond current computational capabilities [@pop2008], necessitating the use of mathematical simplifications and computational shortcuts to approximate a solution [@baker2012].

The key algorithmic approaches to assembly from short read data draw heavily on graph theory, representing overlaps between reads as graphs with overlaps as nodes and the sequences themselves edges, and using graph traversal to trace the path of the original sequence. The de Bruijn graph [@debruijn1946; @idury1995], in which reads are treated as collections of sub-words, is the most commmonly used data structure for this purpose (e.g. @pevzner2001, @zerbino2008, @simpson2009, @robertson2010, @grabherr2011, @schulz2012). Recent theoretical improvements to this approach have been suggested. Using string graphs which do not use sub-words [@myers2005; @simpson2010; @simpson2011; @gonnella2012] or variations on paired de Bruijn graphs which retain read-pairing information [@medvedev2011; @pham2012] are examples, although implementations of these are in their infancy. The only graph traversal algorithms possible for any of these graphs involve the Eulerian Path, in which every edge is visited; solving the alternative Hamiltonian path, visiting every node, is an NP (nondeterministic polynomial time) complete problem [@pevzner1989; @hoogeveen1991], meaning it is not computationally feasible for any significant size of input data. The precise approach differs between genome and transcriptome assembly, because in eukaryotes the transcriptome carries the additional complexity of resolving splice variants.

For all graph assembly methods, errors in read sequences lead to bubbles in the graph, with each error creating two transcripts where only one is real [@baker2012]. Additionally, repetitive sequences can lead to artifical loops which join fragments together *in silico* where so such connection exists *in vivo*, leading to chimeric transcript sequences [@pop2008; @phillippy2008]. Varying the parameters of the graph traversal algorithm can reduce the number of erroneous paths traced, for example by using longer substrings to calculate overlaps and only traversing paths supported by a certain number of connections [@compeau2011]. However, this approach leads to greater fragmentation of the graph as some real corrections are ignored. Thus, all current assembly algorithms can either operate only on very small datasets, or introduce artefacts and errors during the assembly process.

Post-assembly, attempts can be made to remedy the problems introduced at the assembly stage. Current approaches to this problem use mutliple alignment based clustering [@li2006; @edgar2010] to remove redundant contigs representing parts of the same transcript derived from bubbling of the graph. The problems of identifying and removing chimeric transcripts and sequences derived entirely from assembly artefacts are currently unsolved.

There are thus several stages of a <i>de-novo</i> assembly, with multiple programs at each stage, and each program can be configured in various ways. Pipeline development is hampered by the lack of specialised tools for assessing transcriptome assembly quality. While such tools have been developed for genome assembly [@earl2011; @gurevich2013], the same metrics do not provide useful information about transcriptomes.

Assembly tools are non-trivial to operate, often requiring learning a novel interface and performing file-type conversions on the input and output. Moreover, determining the optimal pipeline to use is an open problem, largely because of the complexity of options. There are at least 10 possible programs to include at each of three stages in a pipeline, with at least 20 settings per program. Testing 4 values for each setting would mean testing $(10\times20\times4)^3 = 512,000,000$ possible pipelines. With the running time of assemblers varying from ~20min to >6 hours in this study, clearly it is unfeasible to test all possible pipelines. For this reason, only the most rudimentary optimisation of assembly, in which a few parameters are varied for a single program, has been attempted in previous studies [@chikhi2013].

In summary, there are still significant challenges associated with obtaining highly accurate transcriptome assemblies. In order to address these problems, one research track in this study involves developing software to solve the problems of assessing transcriptome assembly quality, easily developing assembly pipelines, and optimising transcriptome assembly.
